crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -71,23 +70,26 @@ impl DynamicScene {
     ) -> Result<(), SceneSpawnError> {
         let type_registry = type_registry.read();
 
-        // For each component types that reference other entities, we keep track
-        // of which entities in the scene use that component.
-        // This is so we can update the scene-internal references to references
-        // of the actual entities in the world.
-        let mut scene_mappings: TypeIdMap<Vec<Entity>> = Default::default();
-
+        // First ensure that every entity in the scene has a corresponding world
+        // entity in the entity map.
         for scene_entity in &self.entities {
             // Fetch the entity with the given entity id from the `entity_map`
             // or spawn a new entity with a transiently unique id if there is
             // no corresponding entry.
-            let entity = *entity_map
+            entity_map
                 .entry(scene_entity.entity)
                 .or_insert_with(|| world.spawn_empty().id());
-            let entity_mut = &mut world.entity_mut(entity);
+        }
+
+        for scene_entity in &self.entities {
+            // Fetch the entity with the given entity id from the `entity_map`.
+            let entity = *entity_map
+                .get(&scene_entity.entity)
+                .expect("should have previously spawned an empty entity");
```

Comment:
I'm not 100% on Bevy docs style, are we meant to declare these panics in the function docs?

Reply:
Yes, it would be nice to add a `#Panics` section to the docs.

Reply:
My understanding is that panics only need to be documented if they can be triggered by the caller due to invariants not being upheld. That isn't the case for this panic as it signals a bug in the method internals.